src/riscv/lib/src/machine_state/page_cache/dispatch.rs

```diff
@@ -173,19 +173,18 @@ impl<MC: MemoryConfig> DispatchCompiler<MC> for InlineCompiler {
     }
 
     fn compile(target: &JittedPage<Self, MC>, program_counter: Address) -> DispatchFn<Self, MC> {
-        let instr = Jitted::compilation_request_instructions(&target.entries, program_counter);
-
         // The `InlineCompiler` is exclusively used in a single threaded context (compilation is
         // done in the same thread as execution). Therefore, this borrow should never panic as
         // there can be no other attempts to borrow concurrently.
         let mut jit = target.compiler.jit.borrow_mut();
 
-        let fun: DispatchFn<Self, MC> = match jit.compile(&instr, program_counter) {
+        let offset = memory::address_to_page_offset(program_counter) >> 1;
```

Comment:
I'll have a look - but I think it's likely time we switched to moving this into a specific method, it happens a lot